src/components/content/catalog/services/tree/ServiceTree.tsx

```diff
@@ -18,15 +20,60 @@ export function ServiceTree({
     setSelectKey: (selectedKey: React.Key) => void;
     isViewDisabled: boolean;
 }): React.JSX.Element {
-    function isParentTreeSelected(selectKey: React.Key): boolean {
+    const [searchValue, setSearchValue] = useState('');
+
+    const filteredTreeData = useMemo(() => {
```

Comment:
dont know if we really need a useMemo here. The filteredTreeData will anways change whenever searchValue changes. 
Instead we can simply have it a direction function. It will be automatically executed whenever searchValue state changes.